[debug] [2024-03-27T14:11:09.687Z] ----------------------------------------------------------------------
[debug] [2024-03-27T14:11:09.690Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\Thanaporn\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy
[debug] [2024-03-27T14:11:09.690Z] CLI Version:   13.4.0
[debug] [2024-03-27T14:11:09.690Z] Platform:      win32
[debug] [2024-03-27T14:11:09.690Z] Node Version:  v21.6.2
[debug] [2024-03-27T14:11:09.690Z] Time:          Wed Mar 27 2024 21:11:09 GMT+0700 (Indochina Time)
[debug] [2024-03-27T14:11:09.691Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-03-27T14:11:09.750Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-03-27T14:11:09.750Z] > authorizing via signed-in user (hlab.dev.t@gmail.com)
[debug] [2024-03-27T14:11:09.750Z] [iam] checking project human-error-ex for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-03-27T14:11:09.751Z] > refreshing access token with scopes: []
[debug] [2024-03-27T14:11:09.753Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-03-27T14:11:09.753Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-27T14:11:09.797Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------528688264837106378915309","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-03-27T14:11:09.798Z] [iam] error while checking permissions, command may fail: FirebaseError: Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[debug] [2024-03-27T14:11:09.798Z] > refreshing access token with scopes: []
[debug] [2024-03-27T14:11:09.799Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-03-27T14:11:09.799Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-27T14:11:09.803Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------801300032888110468777093","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-03-27T14:11:09.803Z] Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[error] 
[error] Error: Assertion failed: resolving hosting target of a site with no site name or target name. This should have caused an error earlier
[debug] [2024-03-27T14:11:25.265Z] ----------------------------------------------------------------------
[debug] [2024-03-27T14:11:25.267Z] Command:       C:\Program Files\nodejs\node.exe C:\Users\Thanaporn\AppData\Roaming\npm\node_modules\firebase-tools\lib\bin\firebase.js deploy
[debug] [2024-03-27T14:11:25.267Z] CLI Version:   13.4.0
[debug] [2024-03-27T14:11:25.267Z] Platform:      win32
[debug] [2024-03-27T14:11:25.267Z] Node Version:  v21.6.2
[debug] [2024-03-27T14:11:25.268Z] Time:          Wed Mar 27 2024 21:11:25 GMT+0700 (Indochina Time)
[debug] [2024-03-27T14:11:25.268Z] ----------------------------------------------------------------------
[debug] 
[debug] [2024-03-27T14:11:25.315Z] > command requires scopes: ["email","openid","https://www.googleapis.com/auth/cloudplatformprojects.readonly","https://www.googleapis.com/auth/firebase","https://www.googleapis.com/auth/cloud-platform"]
[debug] [2024-03-27T14:11:25.316Z] > authorizing via signed-in user (hlab.dev.t@gmail.com)
[debug] [2024-03-27T14:11:25.316Z] [iam] checking project human-error-ex for permissions ["firebase.projects.get","firebasehosting.sites.update"]
[debug] [2024-03-27T14:11:25.317Z] > refreshing access token with scopes: []
[debug] [2024-03-27T14:11:25.318Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-03-27T14:11:25.318Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-27T14:11:25.357Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------354749753539538878811548","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-03-27T14:11:25.358Z] [iam] error while checking permissions, command may fail: FirebaseError: Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[debug] [2024-03-27T14:11:25.358Z] > refreshing access token with scopes: []
[debug] [2024-03-27T14:11:25.359Z] >>> [apiv2][query] POST https://www.googleapis.com/oauth2/v3/token [none]
[debug] [2024-03-27T14:11:25.359Z] >>> [apiv2][body] POST https://www.googleapis.com/oauth2/v3/token [omitted]
[debug] [2024-03-27T14:11:25.365Z] *** [apiv2] error from fetch(https://www.googleapis.com/oauth2/v3/token, {"headers":{},"method":"POST","body":{"_overheadLength":545,"_valueLength":212,"_valuesToMeasure":[],"writable":false,"readable":true,"dataSize":0,"maxDataSize":2097152,"pauseStreams":true,"_released":true,"_streams":[],"_currentStream":null,"_insideLoop":false,"_pendingNext":false,"_boundary":"--------------------------667454368389609540924506","_events":{},"_eventsCount":1}}): FetchError: request to https://www.googleapis.com/oauth2/v3/token failed, reason: getaddrinfo ENOTFOUND www.googleapis.com
[debug] [2024-03-27T14:11:25.365Z] Authentication Error: Your credentials are no longer valid. Please run firebase login --reauth

For CI servers and headless environments, generate a new token with firebase login:ci
[error] 
[error] Error: Assertion failed: resolving hosting target of a site with no site name or target name. This should have caused an error earlier
